validator/project_validator.go

```diff
@@ -1403,30 +1404,44 @@ func checkTaskRuns(project *model.Project) ValidationErrors {
 						bvtu.Name, bvtu.Variant),
 				})
 			}
+			for _, requester := range bvtu.AllowedRequesters {
+				if requester.Validate() != nil {
+					errs = append(errs, ValidationError{
+						Level: Warning,
+						Message: fmt.Sprintf("task '%s' in build variant '%s' specifies invalid allowed_requester '%s'",
+							bvtu.Name, bvtu.Variant, requester),
+					})
+				} else {
+					hasValidAllowedRequester = true
+				}
+			}
+		} else {
+			hasValidAllowedRequester = true
 		}
-		if bvtu.SkipOnPatchBuild() && bvtu.SkipOnNonPatchBuild() {
+
+		if hasValidAllowedRequester && bvtu.SkipOnPatchBuild() && bvtu.SkipOnNonPatchBuild() {
```

Comment:
Since these all are contingent on `hasValidAllowedRequester` we can call `if hasValidAllowedRequester { // old if statements here }` instead

Reply:
Done.